types/aws-lambda/trigger/appsync-resolver.d.ts

```diff
@@ -1,20 +1,44 @@
 import { Handler } from '../handler';
 
-export type AppSyncResolverHandler<TArguments, TResults, TSource = Record<string, any> | null> = Handler<AppSyncResolverEvent<TArguments, TSource>, TResults | TResults[]>;
+export type AppSyncResolverHandler<TArguments, TResults, TSource = Record<string, any> | null> = Handler<
+    AppSyncResolverEvent<TArguments, TSource>,
+    TResults
+>;
+
+// https:docs.aws.amazon.com/appsync/latest/devguide/tutorial-lambda-resolvers.html#advanced-use-case-batching
+export type AppSyncBatchResolverHandler<TArguments, TResults, TSource = Record<string, any> | null> = Handler<
+    AppSyncResolverEvent<TArguments, TSource>[],
+    TResults[]
+>;
+
+// See https://docs.aws.amazon.com/appsync/latest/devguide/security-authz.html#aws-lambda-authorization
+export type AppSyncAuthorizerHander<TResolverContext = undefined> = Handler<
+    AppSyncAuthorizerEvent,
+    AppSyncAuthorizerResult<TResolverContext>
+>;
 
 export interface AppSyncResolverEventHeaders {
     [name: string]: string | undefined;
 }
 
+export type AppSyncIdentity =
+    | AppSyncIdentityIAM
+    | AppSyncIdentityCognito
+    | AppSyncIdentityOIDC
+    | AppSyncIdentityLambda
+    | undefined
+    | null;
+
 /**
  * See https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html
  *
  * @param TArguments type of the arguments
  * @param TSource type of the source
+ * @param TAuthorizer type of the Authorizer
```

Comment:
You've added a doc for the generic parameter but not the parameter? I think it would improve the tests and usage significantly.

Reply:
I added it, then removed it. It's a bit more complex than it looks.

I agree it would improve usage but it requires more investigation/testing and affects many other things, so I was planning on opening another PR later.

For now, I'll remove the comment.